tests/MeilisearchConsoleCommandTest.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Meilisearch\Scout\Tests;
+
+use Laravel\Scout\EngineManager;
+use MeiliSearch\Client;
+use MeiliSearch\Endpoints\Indexes;
+use MeiliSearch\Exceptions\HTTPRequestException;
+use Meilisearch\Scout\Engines\MeilisearchEngine;
+use Mockery as m;
+
+class MeilisearchConsoleCommandTest extends TestCase
+{
+    /** @test */
+    public function nameArgumentIsRequired()
+    {
+        $this->expectExceptionMessage('Not enough arguments (missing: "name").');
+        $this->artisan('scout:index')
+            ->execute();
+    }
+
+    /** @test */
+    public function commandCreatesIndex()
+    {
+        $client = $this->mock(Client::class);
+        $client->expects('createIndex')->with($indexName = 'testindex', [])->andReturn(m::mock(Indexes::class));
+
+        $engineManager = $this->mock(EngineManager::class);
+        $engineManager->shouldReceive('engine')->with('meilisearch')->andReturn(new MeilisearchEngine(
+            $client
+        ));
+
+        $this->artisan('scout:index', [
+            'name' => $indexName,
+        ])
+            ->expectsOutput('Index "'.$indexName.'" created.')
+            ->assertExitCode(0)
+            ->run();
+    }
+
+    /** @test */
+    public function keyParameterSetsPrimaryKeyOption()
+    {
+        $client = $this->mock(Client::class);
+        $client
+            ->expects('createIndex')
+            ->with($indexName = 'testindex', ['primaryKey' => $testPrimaryKey = 'foobar'])
+            ->andReturn(m::mock(Indexes::class));
+
+        $engineManager = $this->mock(EngineManager::class);
+        $engineManager->shouldReceive('engine')->with('meilisearch')->andReturn(new MeilisearchEngine(
+            $client
+        ));
+
+        $this->artisan('scout:index', [
+            'name' => $indexName,
+            '--key' => $testPrimaryKey,
+        ])
+            ->expectsOutput('Index "'.$indexName.'" created.')
+            ->assertExitCode(0)
+            ->run();
+    }
+
+    /** @test */
+    public function deleteParameterDeletesIndex()
+    {
+        $client = $this->mock(Client::class);
+        $client->expects('deleteIndex')->with($indexName = 'testindex')->andReturn([]);
+
+        $engineManager = $this->mock(EngineManager::class);
+        $engineManager->shouldReceive('engine')->with('meilisearch')->andReturn(new MeilisearchEngine(
+            $client
+        ));
+
+        $this->artisan('scout:index', [
+            'name' => $indexName,
+            '--delete' => true,
+        ])
+            ->expectsOutput('Index "'.$indexName.'" deleted.')
+            ->assertExitCode(0)
+            ->run();
+    }
+
+    /** @test */
+    public function commandReturnsErrorStatusCodeOnException()
+    {
+        /**
+         * This somehow can't assert that the exception exception is actually thrown.
+         * The codepath the the catch block in the command is executed.
+         */
+        // $this->expectException(HTTPRequestException::class);
```

Comment:
@shokme or @curquiza maybe you know some more details on how to check this

Reply:
Sorry, I might miss understood, I'm not used to mocking, but isn't the line below enough?
```php
$client->expects('createIndex')->andThrow(new HTTPRequestException(404, ['message' => 'Testmessage']));
```
What do you want to do exactly?

Reply:
You are right. I had some misunderstanding of the try catch block in the command. The exception is handled and therefore we can't expect the exception in the test.

I removed it.